### PR TITLE
Validates user input

### PR DESCRIPTION
**Pull request by amprokop**
Shows error message to users if Veteran ID is invalid.

Connects https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/496 , https://github.com/department-of-veterans-affairs/caseflow-efolder/issues/516

efolder_error_message

### CODE REVIEW

**Comment by shellicious** [comment]
@amprokop:

Header text in the error should say: "The veteran ID you entered was not valid."
Body text in the header should say: "Check to make sure you entered the ID correctly and try again."

Are there specific reasons we can give to help the user enter a valid ID? No special characters, numbers instead of letters?

File: Gemfile [hunk 1]
```diff
@@ -112,6 +112,11 @@ end
 group :development do
   # Access an IRB console on exception pages or by using <%= console %> in views
   #gem 'web-console', '~> 2.0'
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: :ruby
```
Comment (amprokop) [inline Gemfile]: adding byebug since a lot of devs use it

Comment (aroltsch) [inline Gemfile]: fine! I like pry :)

File: app/models/search.rb [hunk 2]
```diff
@@ -14,6 +14,19 @@ class Search < ActiveRecord::Base
...
 
+  def valid_file_number?
+    number = sanitized_file_number
+    return true if number =~ /DEMO/
+
+    return false if /\D+/ =~ number
+    # We don't want to render an error message on initial pageload.
+    # We don't pass in user until we actually submit the form,
+    # so only check length if the user attribute is present.
+    return false if user && number.length < 8
```
Comment (aroltsch) [inline app/models/search.rb]: Would this work?
```
return false if /\D+/ =~ number ||  (user && number.length < 8)
```

File: app/models/search.rb [hunk 3]
```diff
@@ -14,6 +14,19 @@ class Search < ActiveRecord::Base
     access_denied: 3
   }
 
+  def valid_file_number?
```
Comment (aroltsch) [inline app/models/search.rb]: Is there a unit test for this?

File: app/models/search.rb (hunk 2 above)
Comment (amprokop) [inline app/models/search.rb]: sure! that sounds fine. i thought it was a bit clearer to break it out into two conditions so i can explain the second one. happy to crunch it down into one though

Comment (aroltsch) [inline app/models/search.rb]: Oh okay, I am gonna leave it up to you then.

File: app/models/search.rb (hunk 3 above)
Comment (amprokop) [inline app/models/search.rb]: Ah I'll write one

Comment (aroltsch) [inline app/models/search.rb]: sorry Alex :)

**Pull request merged**